src/code_1AC760.c

```diff
@@ -42,10 +143,49 @@ INCLUDE_ASM(s32, "code_1AC760", PartnerPowerBounceEnemy);
 
 INCLUDE_ASM(s32, "code_1AC760", PartnerTestEnemy);
 
-INCLUDE_ASM(s32, "code_1AC760", func_8028070C);
+ApiStatus func_8028070C(ScriptInstance* script, s32 isInitialCall) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    Bytecode* args = script->ptrReadPos;
+    Actor* actor = get_actor(script->owner1.actorID);
+    s32 damageAmount = get_variable(script, *args++);
+    s32 event = get_variable(script, *args++);
+    
+    battleStatus->currentTargetID = actor->targetActorID;
+    battleStatus->currentTargetPart = actor->targetPartIndex;
+
+    if (dispatch_damage_event_partner_0(damageAmount, event, battleStatus) >= 0) {
+        return does_script_exist_by_ref(script) ? ApiStatus_DONE2 : ApiStatus_BLOCK;
```

Comment:
I'd rather use not ternaries if possible

Reply:
I disagree, unless the `if else` version doesn't match.
IIRC this was a classic `(bool != 0) * 2` thing we see fairly often - and we should change the other ones to look like this / `if return else return` if that matches.

Reply:
discussed in chat and I agree that in this case they make sense